File: cpp/Aksesoris.cpp
#pragma once
#include "PetShop.cpp"

using namespace std;

class Aksesoris : public PetShop
{
protected:
    string bahan;
    string warna;

public:
    Aksesoris() : PetShop(), bahan(""), warna("") {}
    Aksesoris(int id, string nama_produk, int harga_produk, int stok_produk, string kategori_produk, string bahan, string warna)
        : PetShop(id, nama_produk, harga_produk, stok_produk, kategori_produk), bahan(bahan), warna(warna) {}

    void set_bahan(string bahan) { this->bahan = bahan; }
    void set_warna(string warna) { this->warna = warna; }

    string get_bahan() const { return bahan; }
    string get_warna() const { return warna; }

    ~Aksesoris() {}
};

File: php/Aksesoris.php
<?php

require_once 'PetShop.php';

class Aksesoris extends PetShop
{ // mendefinisikan kelas Aksesoris yang merupakan turunan dari PetShop
    protected $bahan; // atribut untuk menyimpan bahan aksesoris
    protected $warna; // atribut untuk menyimpan warna aksesoris

    // konstruktor default, menginisialisasi atribut dengan nilai default
    public function __construct($id = 0, $nama_produk = "", $harga_produk = 0, $stok_produk = 0, $kategori_produk = "", $foto = "", $bahan = "", $warna = "")
    {
        parent::__construct($id, $nama_produk, $harga_produk, $stok_produk, $kategori_produk, $foto);
        $this->bahan = $bahan;
        $this->warna = $warna;
    }

    // metode setter untuk atribut bahan
    public function setBahan($bahan)
    {
        $this->bahan = $bahan;
    }

    // metode setter untuk atribut warna
    public function setWarna($warna)
    {
        $this->warna = $warna;
    }

    // metode getter untuk atribut bahan
    public function getBahan()
    {
        return $this->bahan;
    }

    // metode getter untuk atribut warna
    public function getWarna()
    {
        return $this->warna;
    }
}
?>